script/packagecloud.rb

```diff
@@ -114,7 +114,9 @@ def distro_names_for(filename)
     puts "pushing #{full_path} to #{$distro_id_map.key(distro_id).inspect}"
     result = $client.put_package("git-lfs", pkg, distro_id)
     result.succeeded || begin
-      raise "packagecloud put_package failed, error: #{result.response}"
+      if result.response != '{"filename":["has already been taken"]}'
```

Comment:
Could you give some additional context on why this if-statement is necessary? I don't know much about packagecloud, so I'm still a bit confused after reading the commit message.

Reply:
Sure. During the release process, the upload aborted halfway through due to `sles/12.4` not existing. Re-running the script causes us to attempt to re-upload the same files. If the file already exists and this conditional isn't present, PackageCloud returns this error and we throw an exception and abort the upload process, since PackageCloud doesn't allow us to upload over a file with the same name.

Instead, the thing we want to do is just ignore the fact that we've already uploaded those files and continue the process. Similarly, when we build the CentOS 6 packages, we always build the same Ruby and ronn packages, and without this change, they have to be manually deleted from the directory or the script will abort when the upload gets to them.

Hopefully this makes things a bit clearer.

Reply:
Ah okay! So we can still upload the package if we receive this "package already exists" error, we just weren't doing it before? Could you add a comment here to that effect?

Reply:
Well, not exactly. This error means we've already uploaded the package, and we can't upload it a second time, so we're ignoring the failure when we attempt to do so.

I'll add an appropriate comment.